src/common-model/functions.ts

```diff
@@ -134,39 +137,64 @@ export function getClinicalEntitySubmittedData(
 ): ClinicalInfo[] {
   const result = getClinicalObjectsFromDonor(donor, clinicalEntitySchemaName) as any[];
 
-  const clinicalRecords =
-    clinicalEntitySchemaName === ClinicalEntitySchemaNames.DONOR
-      ? result.map((entity: any) => ({
+  let clinicalRecords = [];
+
+  switch (clinicalEntitySchemaName) {
+    case ClinicalEntitySchemaNames.DONOR: {
+      clinicalRecords = result.map((entity: any) => ({
+        donor_id: donor.donorId,
+        program_id: donor.programId,
+        ...entity.clinicalInfo,
+      }));
+      break;
+    }
+    case ClinicalEntitySchemaNames.TREATMENT: {
+      clinicalRecords = result.map((treatment: any) => {
+        const clinicalInfo = treatment.clinicalInfo || {};
+        const therapy_type =
+          treatment.therapies.length === 1
+            ? { therapy_type: treatment.therapies[0].therapyType }
+            : {};
+        const therapy_info =
+          treatment.therapies.length === 1 && treatment.therapies[0].clinicalInfo;
```

Comment:
checking for `treatment.therapies.length` and dipping into array twice here.
 
 - Will `treatment.therapies` at least be an empty array?
 - cleaner way to do it? maybe something like...
 ```
 const therapy = treatment.therapies && treatment.therapies.length > 0 ?  treatment.therapies[0] : {} ;
 const {therapyTypes = { }, clinicalInfo = {}} = therapy;
 
 ```

Reply:
I've added Types here + yes treatment.therapies should always be an array

Reply:
Agree w/ feedback but need to give this a more thorough review b/c it relates to `Therapy` and the ticket is focused on `Surgery` bug, need to limit to refactoring existing logic
I'm questioning if/how this logic should be changed but that's out of scope to ticket

Reply:
Just a cleaner (imho) way of writing the same logic that's been refactored so the same conditional isn't being checked and the same array item is being pulled out. No worries if you want to leave it